dali/operators/reader/loader/webdataset/tar_utils.cc

```diff
@@ -97,11 +98,11 @@ static tartype_t kTarArchiveType = {LibtarOpenTarArchive, [](int) -> int { retur
 
 TarArchive::TarArchive(std::unique_ptr<FileStream> stream_)
     : stream_(std::move(stream_)),
-      archiveoffset_(0),
       instance_handle_(Register(this)),
-      eof_(false) {
+      archiveoffset_(0) {
```

Comment:
I believe @mzient's comment about members in the initializer list may be also applied here?

Reply:
well this one is initialized with a constant value, the issue with the other one was that it was dependent on the size of this->stream_

Reply:
When initializing with a constant, it's preferable to use in-class initialization - this helps avoid uninitialized members when adding new constructors.